.circleci/config.yml

```
@@ -138,6 +138,132 @@ jobs:
+            for repo in quay.io ghcr.io; do 
+              tag=$(cat ${dir}/TAG)
+              c="$(container-tag-exists ${repo}/cybozu/$image $tag)"
+              if [ "$c" = "" ]; then
+                  echo ${repo}/cybozu/$image >> BUILDS
+              fi
+            done
+      - run:
+          name: Validate consistency between BRANCH and TAG
+          command: |
+            dir=<< parameters.dir >>
+            if [ "$dir" = "" ]; then dir=<< parameters.container-image >> ; fi
+            if [ -e "$dir/NO_TAG_BRANCH_CONSISTENCY" ]; then exit 0; fi
+            ./tag_branch_consistency $dir
+      - run:
+          name: Build test
+          command: |
+            if [ "${CIRCLE_BRANCH}" == "main" ]; then
+                exit 0
+            fi
+            if [ ! -f BUILDS ]; then
+                exit 0
+            fi
+            echo "building test: << parameters.container-image >> ..."
+            image=<< parameters.container-image >>
+            dir=<< parameters.dir >>
+            if [ "$dir" = "" ]; then dir=<< parameters.container-image >> ; fi
+            before=<< parameters.before >>
+            cd $dir
+            bash -c "$before"
+            docker context create buildx
+            docker buildx create buildx --use
+            docker buildx build \
+              --platform linux/amd64,linux/arm64/v8 \
+              -o type=tar,dest=$image.tar \
+              .
+      - run:
+          name: Build and Push
+          command: |
+            if [ "${CIRCLE_BRANCH}" != "main" ]; then
+                exit 0
+            fi
+            if [ ! -f BUILDS ]; then
+                exit 0
+            fi
+            docker login -u $QUAY_USER -p $QUAY_PASSWORD quay.io
+            docker login -u $GITHUB_USER -o $GITHUB_TOKEN ghcr.io
+            echo
+            echo "building and pushing << parameters.container-image >> ..."
+            image=<< parameters.container-image >>
+            dir=<< parameters.dir >>
+            if [ "$dir" = "" ]; then dir=<< parameters.container-image >> ; fi
+            TAG=$(cat $dir/TAG)
+            BRANCH=$(cat $dir/BRANCH)
+            before=<< parameters.before >>
+            cd $dir
+            bash -c "$before"
+            docker context create buildx
+            docker buildx create buildx --use
+            for repo in $(cat BUILDS) ; do
+              docker buildx build \
+                --push \
+                --platform linux/amd64,linux/arm64/v8 \
+                -t ${repo}:$TAG
+                .
+            done
+
+            if echo $TAG | grep -q -e - ; then
+              echo ===== Skip pushing branch tags for pre-release $TAG =====
+              exit 0
+            fi
+            if [ -f $dir/BRANCH ]; then
+                BRANCH=$(cat $dir/BRANCH)
+                for repo in $(cat BUILDS) ; do
+                  docker buildx imagetools create --tag ${repo}/cybozu/$image:$TAG ${repo}/cybozu/$image:$BRANCH
```

You may need to reverse the arguments.

```suggestion
                  docker buildx imagetools create --tag ${repo}/cybozu/$image:$BRANCH ${repo}/cybozu/$image:$TAG
```